cmd/soroban-cli/src/events.rs

```diff
@@ -0,0 +1,191 @@
+use std::io;
+use std::io::Write;
+
+use clap::Parser;
+use hex::FromHexError;
+use soroban_env_host::xdr::{ReadXdr, ScVal};
+use termcolor::{Color, ColorChoice, ColorSpec, StandardStream, WriteColor};
+
+use crate::rpc::Event;
+use crate::utils;
+use crate::{rpc, rpc::Client};
+use crate::{HEADING_RPC, HEADING_SANDBOX};
+
+#[derive(Parser, Debug)]
+#[clap()]
+pub struct Cmd {
+    /// The ledger range to pull events from
+    start_ledger: u32,
+    end_ledger: u32,
```

Comment:
:thought_balloon: The start and end ledger will be the first place in the CLI that there are positional arguments. Every existing command has only option arguments. I don't think we need to be religious about it at this point, there might be commands where there is a definite primary participant who could be the positional argument. In this case, these values don't feel like are that though. End ledger feels like it should be optional at the least.

Reply:
I knew this comment was coming :sunglasses: I adopted a philosophy of "optional arguments should be optional" some time ago, so I always defer to using positional arguments for things that are required. The end ledger may become optional, yeah, but the spec needs to be updated first. (I suppose we could make a `getLatestLedger`-esque call as an alternative, but that'd be a post-v0 enhancement imo.) Anyway, like you, I don't want to be religious about it, just make the case for making things that are required a little more explicit. I'm happy to change this to `--start-ledger` and `--end-ledger`, but wanted this thread to happen, first.

Thinking about this a bit, though:

>  there might be commands where there is a definite primary participant who could be the positional argument

it might actually make sense to have `<rpc-url-or-sandbox>` as a positional argument.

Reply:
> it might actually make sense to have `<rpc-url-or-sandbox>` as a positional argument.

Something to consider it's difficult to make positional arguments consistent across different commands. Take the `<rpc-url-or-sandbox>` option. If it was positional, would it be positional on every command? That also implies it can't be provided some other way like an ENV var.

> I adopted a philosophy of "optional arguments should be optional" some time ago

This makes for some awkward commands. For example, if end ledger is optional and start ledger is mandatory, it's quite difficult to understand just by reading the following command what it does.

```
soroban events --end-ledger 20 15
```

Reply:
That's true! Though you'd write `events 15 --end-ledger 20` for clarity. But still awkward. Your note about envvars is very compelling, too :+1: anyway, fixed in latest commit